lib/collection/src/shards/local_shard/mod.rs

```diff
@@ -990,53 +990,49 @@ impl LocalShard {
     }
 
     pub async fn local_shard_status(&self) -> (ShardStatus, OptimizersStatus) {
-        let mut status = ShardStatus::Green;
-        let mut optimizer_status = OptimizersStatus::Ok;
-
         {
             let segments = self.segments().read();
 
+            // Red status on failed operation or optimizer error
             if !segments.failed_operation.is_empty() || segments.optimizer_errors.is_some() {
-                status = ShardStatus::Red;
+                let optimizer_status = segments
+                    .optimizer_errors
+                    .as_ref()
+                    .map(|err| OptimizersStatus::Error(err.to_string()))
+                    .unwrap_or_default();
```

Comment:
IMO, `map_or` would be easier to read. I could only understand what's going on here by looking at previous impl and seeing that "default" `optimizer_status` is `OptimizersStatus::Ok`.

```suggestion
                    .map_or(OptimizersStatus::Ok, |err| OptimizersStatus::Error(err.to_string()));
```

Reply:
Sure! :+1:

Reply:
Btw, I'm not sure if that would pass CI checks or compile, but thanks for the confidence 😂

Reply:
It doesn't :face_holding_back_tears: